_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -1041,11 +1043,76 @@ private function _process_onboarding( $data ) {
 			}
 		}
 
+		// Add/update business address widget
+		if ( isset( $data['businessAddress'] ) ) {
+			//grab the first sidebar and proceed if it is present
+			$first_sidebar = self::get_first_sidebar();
+
+			if( $first_sidebar ) {
+				$title = wp_unslash( $data[ 'businessAddress' ][ 'name' ] );
+				$address = wp_unslash(
+					$data[ 'businessAddress' ][ 'city' ] . ' ' .
+					$data[ 'businessAddress' ][ 'state' ] . ' ' .
+					$data[ 'businessAddress' ][ 'street' ] . ' ' .
+					$data[ 'businessAddress' ][ 'zip' ]
+				);
+				$widget_options = array(
+					'title'   => $title,
+					'address' => $address,
+					'phone'   => '',
+					'hours'   => '',
+					'showmap' => false,
+					'email' => '',
+					'apikey' => '',
+				);
+				$position = '0';
+				$id_base = 'widget_contact_info';
+
+				$has_ba_widget = self::has_business_info_widget( $first_sidebar );
+				if ( $has_ba_widget ) {
+					self::update_widget( $id_base, $first_sidebar, $position, $widget_options);
+				}	else {
+					Jetpack_Widgets::activate_widget( $id_base, $first_sidebar, $position, $widget_options);
+				}
+			}
+		}
+
 		return empty( $error )
 			? ''
 			: join( ', ', $error );
 	}
 
+		public function get_first_sidebar() {
+			$active_sidebars = Jetpack_Widgets::get_active_sidebars();
+
+			unset( $active_sidebars[ 'wp_inactive_widgets' ], $active_sidebars[ 'array_version' ] );
+
+			if ( empty( $active_sidebars ) ) {
+				return false;
+			}
+			return array_shift( array_keys( $active_sidebars ) );
+		}
+
+		public function has_business_info_widget( $sidebar ) {
```

Comment:
Can we make this a method in `Jetpack_Widgets` that accepts a widget base ID (`widget_contact_info` in our case)? Could be useful, and would be the right place to keep this code IMHO.
 
Also, can we use something from `Jetpack_Widgets` to make this one simpler? `get_widget_by_id_base()` looks like a good candidate.

Reply:
@tyxla, I see the point, and at the same time thinking that it is pretty much a JPO-specific method. Will it be reusable by being defined in a Jetpack module? 🤔

Reply:
No usages right now AFAIK, but it doesn't cost anything (other than just adding a parameter for the widget ID base) to do it, and in case we decide to use it in the future, it will be easier and will not require refactor PRs like this one #8464.